src/node_file.cc

```diff
@@ -1079,41 +1081,101 @@ static void InternalModuleReadJSON(const FunctionCallbackInfo<Value>& args) {
   }
 
   const size_t size = offset - start;
-  char* p = &chars[start];
-  char* pe = &chars[size];
-  char* pos[2];
-  char** ppos = &pos[0];
-
-  while (p < pe) {
-    char c = *p++;
-    if (c == '\\' && p < pe && *p == '"') p++;
-    if (c != '"') continue;
-    *ppos++ = p;
-    if (ppos < &pos[2]) continue;
-    ppos = &pos[0];
-
-    char* s = &pos[0][0];
-    char* se = &pos[1][-1];  // Exclude quote.
-    size_t n = se - s;
-
-    if (n == 4) {
-      if (0 == memcmp(s, "main", 4)) break;
-      if (0 == memcmp(s, "name", 4)) break;
-      if (0 == memcmp(s, "type", 4)) break;
-    } else if (n == 7) {
-      if (0 == memcmp(s, "exports", 7)) break;
-      if (0 == memcmp(s, "imports", 7)) break;
+  simdjson::ondemand::parser parser;
+  auto document = parser.iterate(&chars[start], size, sizeof(&chars[start]));
```

Comment:
This is fine but it might potentially throw an exception. If you don't want to mess with exceptions, do...

```
simdjson::ondemand::document doc;
auto err = parser.iterate(json_string).get(doc);
if(err) {
// I have some kind of error here
}
```

Doing this throughout is painful, so instead you can just use a try/catch.

Reply:
You can do...

```C++
simdjson::padded_string json_string(chars + start, size);
simdjson::ondemand::document doc;
try {
  doc = parser.iterate(json_string);
  ....
} catch (simdjson::simdjson_error &error) {
    std::cerr << "JSON error: " << error.what() << " near "
              << doc.current_location() << " in " << json_string << std::endl;
    FAIL();
}
```